### DISCUSSION TITLE
Starting flow

### DISCUSSION THREAD

**Discussion by Artem-Gulyaev** [MariaShishkinaIFZ/geophys#6] [General]
First proposal is to fork the workflow into two parallel streams:

- zeroth branch: cleanup and upload all legacy code which is necessary (WIP by @Artem-Gulyaev for now)
- first branch: bring the old code alive and buildable in std linux env (if @iod-ine is OK to handle this would be cool)
- second branch: define data flows in the system and identify the components and their API (something I would like to participate in, and where help of @MariaShishkinaIFZ  and @Sergey will be needed)

this is it for now. Later there will be more information on the further steps. But these are relatively clear ones.

**Comment by Artem-Gulyaev** [comment]
@iod-ine, FYI probably this will work to notify you... If not, then will wait for @MariaShishkinaIFZ to pull you into the project

**Comment by iod-ine** [comment]
Sounds good, I'll start looking into it as soon as possible.

**Comment by iod-ine** [comment]
What do you think about using Docker for deployment so that we don't have to worry about building within different environments?

**Reply by Artem-Gulyaev** [reply]
Yeap, Docker is good thing for a stuff which needs to run in generic environment, absolutely. Especially if testing is considered, then Docker is the CI support out of the box almost. So for sure it is a good way to go. For now I would use CMake as a build system and Docker on top of it to make system-independent containers. Would this work for you?

**Reply by Artem-Gulyaev** [reply]
PS: first local build on a smallest util shows 50 linkage errors and some compile errors)) there is definitely something to do))

**Reply by Artem-Gulyaev** [reply]
![image](https://user-images.githubusercontent.com/39741342/117498606-50b76800-af7a-11eb-8732-f0f38014629c.png)
))) @iod-ine  I see such syntax first time in my life))) there is always a time to learn something new))

**Reply by iod-ine** [reply]
@Artem-Gulyaev yes, CMake + Docker is exactly what I had in mind :) Can't wait to dive deep into all these errors
